cdc/owner/changefeed.go

```diff
@@ -189,7 +193,12 @@ func (c *changefeed) tick(ctx cdcContext.Context, state *orchestrator.Changefeed
 	default:
 	}
 
-	c.sink.emitCheckpointTs(checkpointTs, c.schema.AllTableNames())
+	// This means that the cached DDL has been executed,
+	// and we need to use the latest table names.
+	if c.currentTableNames == nil {
+		c.currentTableNames = c.schema.AllTableNames()
+	}
+	c.sink.emitCheckpointTs(checkpointTs, c.currentTableNames)
```

Comment:
If a DDL is being executed asynchronously, could the `emitCheckpointTs` run with the same `checkpointTs` but different `c.currentTableNames`?

Reply:
Before the DDL really executes, we need to use the old currentTableNames, otherwise we will have problems with sending early or stopping early.